### DISCUSSION TITLE
Inconsistency of empty state filter between Repository.issues and Repository.pullRequests fields

### DISCUSSION THREAD

**Discussion by dmitshur** [community/community#24849] [Apps, API and Webhooks]
<p>I’ve run into this minor inconsistency in the GitHub GraphQL API v4. Consider the following query:</p>
<pre><code>query {
  repository(owner:"octocat", name:"Hello-World") {
    allIssues: issues {
      totalCount
    }
    allPRs: pullRequests {
      totalCount
    }
    issues (states:[]) {
      totalCount
    }
    pullRequests (states:[]) {
      totalCount
    }
  }
}

{
  "data": {
    "repository": {
      "allIssues": {
        "totalCount": 250
      },
      "allPRs": {
        "totalCount": 249
      },
      "issues": {
        "totalCount": 0
      },
      "pullRequests": {
        "totalCount": 249
      }
    }
  }
}
</code></pre>
<p>Notice that states:[] filters out all issues, but includes all PRs.</p>
<p>This is a minor inconsistency. It’d be better if providing states:[] had the same effect on issues and PRs.</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/dmitshur">@dmitshur</a>,</p>
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>
<p>Cheers!</p>
